content/master/concepts/providers.md

```diff
@@ -3,52 +3,306 @@ title: Providers
+for more information.
+{{< /hint >}}
+
+## Install a Provider
+
+Installing a provider creates a Provider pod that's responsible for installing
+the Provider's APIs into the Kubernetes cluster. Providers constantly watch the
+state of the desired managed resources and create any external resources that
+are missing. 
+
+Install a Provider with a Crossplane
+{{<hover label="install" line="2">}}Provider{{</hover >}} object setting the
+{{<hover label="install" line="6">}}spec.package{{</hover >}} value to the
+location of the provider package.
+
+For example, to install the 
+[AWS Community Provider](https://github.com/crossplane-contrib/provider-aws),
+
+```yaml {label="install"}
 apiVersion: pkg.crossplane.io/v1
 kind: Provider
 metadata:
   name: provider-aws
 spec:
-  package: "xpkg.upbound.io/crossplane-contrib/provider-aws:v0.33.0"
+  package: xpkg.upbound.io/crossplane-contrib/provider-aws:v0.39.0
+```
+
+{{< hint "tip" >}}
+Providers are Crossplane Packages. Read more about Packages in the
+[Packages documentation]({{<ref "packages" >}}).
+{{< /hint >}}
+
+By default, the Provider pod installs in the same namespace as Crossplane
+(`crossplane-system`).
+
+### Install with Helm
+
+Crossplane supports installing Providers during an initial Crossplane 
+installation with the Crossplane Helm chart.
+
+Use the 
+{{<hover label="helm" line="5" >}}--set provider.pacakges{{</hover >}} 
+argument with `helm install`. 
+
+For example, to install the AWS Community Provider,
+
+```shell {label="helm"}
+helm install crossplane \
+crossplane-stable/crossplane \
+--namespace crossplane-system \
+--create-namespace \
+--set provider.packages={xpkg.upbound.io/crossplane-contrib/provider-aws:v0.39.0}
+```
+
+### Install from a private repository
+
+Installing a Provider from a private package repository requires a
+Kubernetes secret object. The Provider uses the secret with the
+{{<hover label="pps" line="7" >}}packagePullSecrets{{</hover>}} option. 
+
+```yaml {label="pps"}
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: private-provider
+spec:
+  package: private-repo.example.org/providers/my-provider
+  packagePullSecrets:
+    - name: my-secret
+```
+## Upgrade a Provider
+
+To upgrade an existing Provider edit the installed Provider Package by either
+applying a new Provider manifest or with `kubectl edit providers`.
+
+Update the version number in the Provider's `spec.package` and apply the change. 
+Crossplane installs the new image and creates a new `ProviderRevision`.
+
+## Remove a Provider
+
+Remove a Provider by deleting the Provider object with `kubectl delete
+provider`. 
+
+{{< hint "warning" >}}
+Removing a Provider without first removing the Provider's managed resources 
+may abandon the resources. The external resources aren't deleted.
+
+You must manually delete external resources through your cloud provider. Managed
+resources must be manually deleted by removing their finalizers. 
```

Comment:
Perhaps prefix with "in this case" or similar. Otherwise this could be read to mean that you must do this in happy path, i.e. that you must always delete external resources via your cloud provider before uninstalling a provider.

Reply:
Added "If you remove the Provider first, you must manually delete..."